src/Sulu/Component/Security/Authorization/SecurityContextVoter.php

```diff
@@ -141,12 +92,92 @@ public function vote(TokenInterface $token, $object, array $attributes)
     }
 
     /**
+     * Checks if the user has the required permissions to access the given object.
+     *
+     * @param SecurityCondition $object
+     * @param UserInterface $user
+     * @param $attribute
+     *
+     * @return bool
+     */
+    private function voteAccessControl(SecurityCondition $object, UserInterface $user, $attribute)
+    {
+        if (!$object->getObjectType() || !$object->getObjectId()) {
+            return;
+        }
+
+        $permissions = $this->accessControlManager->getPermissions($object->getObjectType(), $object->getObjectId());
+
+        if (empty($permissions)) {
+            return;
+        }
+
+        $roles = $user->getUserRoles();
```

Comment:
This method is actually not part of the `UserInterface` and iterating afterwards is not very nice, because I always have to call `getRole` on the `UserRole`. It would be nice to have a `getRoles` method to retrieve all roles of the user directly, but this method is already used by Symfony. Should probably still do that, and think of a better method name.